core/src/main/java/io/kestra/core/validations/ValidationFactory.java

```diff
@@ -183,18 +186,22 @@ ConstraintValidator<FlowValidation, Flow> flowValidation() {
 
             List<String> violations = new ArrayList<>();
 
-            // task unique id
+            // tasks unique id
             List<String> taskIds = value.allTasksWithChilds()
                 .stream()
                 .map(Task::getId)
                 .toList();
-            List<String> taskDuplicates = taskIds
-                .stream()
-                .distinct()
-                .filter(entry -> Collections.frequency(taskIds, entry) > 1)
+
+            List<String> duplicateIds = getDuplicates(taskIds);
+
+            List<String> triggerIds = Optional.ofNullable(value.getTriggers()).map(Collection::stream).orElse(Stream.empty())
+                .map(AbstractTrigger::getId)
                 .toList();
-            if (taskDuplicates.size() > 0) {
-                violations.add("Duplicate task id with name [" + String.join(", ", taskDuplicates) + "]");
+
+            duplicateIds.addAll(getDuplicates(triggerIds));
+
+            if (!duplicateIds.isEmpty()) {
+                violations.add("Duplicate task id with name [" + String.join(", ", duplicateIds) + "]");
```

Comment:
You cannot merge tasks and triggers, they are different things.
For a trigger you expect having a message "Duplicate trigger id with name ..."